StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1412FixAllProvider.cs

```diff
@@ -36,34 +36,34 @@ internal sealed class SA1412FixAllProvider : FixAllProvider
                 return null;
             }
 
-            return CodeAction.Create(MaintainabilityResources.SA1412CodeFix, token => Task.FromResult(newSolution));
+            return CodeAction.Create(string.Format(MaintainabilityResources.SA1412CodeFix, await SA1412CodeFixProvider.GetEncodingNameForDocumentAsync(fixAllContext.Document).ConfigureAwait(false)), token => Task.FromResult(newSolution));
         }
 
-        private static async Task<Solution> GetProjectFixesAsync(FixAllContext fixAllContext, Project project)
+        private static async Task<Solution> FixDocumentAsync(FixAllContext fixAllContext, Document document)
         {
-            Solution solution = project.Solution;
-
-            var documentIds = project.DocumentIds;
+            Solution solution = document.Project.Solution;
+            var diagnostics = await fixAllContext.GetDocumentDiagnosticsAsync(document).ConfigureAwait(false);
 
-            foreach (var documentId in documentIds)
+            if (diagnostics.Length == 0 || fixAllContext.CodeActionEquivalenceKey != await SA1412CodeFixProvider.GetEquivalenceKeyForDocumentAsync(document).ConfigureAwait(false))
             {
-                solution = await FixDocumentAsync(fixAllContext, solution.GetDocument(documentId)).ConfigureAwait(false);
+                return solution;
             }
 
-            return solution;
+            return await SA1412CodeFixProvider.GetTransformedSolutionAsync(document).ConfigureAwait(false);
         }
 
-        private static async Task<Solution> FixDocumentAsync(FixAllContext fixAllContext, Document document)
+        private static async Task<Solution> GetProjectFixesAsync(FixAllContext fixAllContext, Project project)
         {
-            Solution solution = document.Project.Solution;
-            var diagnostics = await fixAllContext.GetDocumentDiagnosticsAsync(document).ConfigureAwait(false);
+            Solution solution = project.Solution;
+
+            var documentIds = project.DocumentIds;
 
-            if (diagnostics.Length == 0)
+            foreach (var documentId in documentIds)
             {
-                return solution;
+                solution = await FixDocumentAsync(fixAllContext, solution.GetDocument(documentId)).ConfigureAwait(false);
```

Comment:
It's possible, but *believe me when I say* the problems with this code fix are not the performance of this block.